src/logger/src/logger.rs

```diff
@@ -1,776 +1,343 @@
-//! The logging destination is specified upon the initialization of the logging system.
-//!
-//! # Enabling logging
-//! There are 2 ways to enable the logging functionality:
-//!
-//! 1) Calling `LOGGER.configure()`. This will enable the logger to work in limited mode.
-//! In this mode the logger can only write messages to stdout or stderr.
-
-//! The logger can be configured in this way any number of times before calling `LOGGER.init()`.
-//!
-//! 2) Calling `LOGGER.init()`. This will enable the logger to work in full mode.
-//! In this mode the logger can write messages to arbitrary buffers.
-//! The logger can be initialized only once. Any call to the `LOGGER.init()` following that will
-//! fail with an explicit error.
-//!
-//! ## Example for logging to stdout/stderr
-//!
-//! ```
-//! use std::ops::Deref;
-//!
-//! use logger::{error, warn, LOGGER};
-//!
-//! // Optionally do an initial configuration for the logger.
-//! if let Err(err) = LOGGER.deref().configure(Some("MY-INSTANCE".to_string())) {
-//!     println!("Could not configure the log subsystem: {}", err);
-//!     return;
-//! }
-//! warn!("this is a warning");
-//! error!("this is an error");
-//! ```
-//! ## Example for logging to a `File`:
-//!
-//! ```
-//! use std::io::Cursor;
-//!
-//! use libc::c_char;
-//! use logger::{error, warn, LOGGER};
-//!
-//! let mut logs = Cursor::new(vec![0; 15]);
-//!
-//! // Initialize the logger to log to a FIFO that was created beforehand.
-//! assert!(LOGGER
-//!     .init("Running Firecracker v.x".to_string(), Box::new(logs),)
-//!     .is_ok());
-//! // The following messages should appear in the in-memory buffer `logs`.
-//! warn!("this is a warning");
-//! error!("this is an error");
-//! ```
-
-//! # Plain log format
-//! The current logging system is built upon the upstream crate 'log' and reexports the macros
-//! provided by it for flushing plain log content. Log messages are printed through the use of five
-//! macros:
-//! * error!(<string>)
-//! * warning!(<string>)
-//! * info!(<string>)
-//! * debug!(<string>)
-//! * trace!(<string>)
-//!
-//! Each call to the desired macro will flush a line of the following format:
-//! ```<timestamp> [<instance_id>:<level>:<file path>:<line number>] <log content>```.
-//! The first component is always the timestamp which has the `%Y-%m-%dT%H:%M:%S.%f` format.
-//! The level will depend on the macro used to flush a line and will be one of the following:
-//! `ERROR`, `WARN`, `INFO`, `DEBUG`, `TRACE`.
-//! The file path and the line provides the exact location of where the call to the macro was made.
-//! ## Example of a log line:
-//! ```bash
-//! 2018-11-07T05:34:25.180751152 [anonymous-instance:ERROR:vmm/src/lib.rs:1173] Failed to write
-//! metrics: Failed to write logs. Error: operation would block
-//! ```
-//! # Limitations
-//! Logs can be flushed either to stdout/stderr or to a byte-oriented sink (File, FIFO, Ring Buffer
-//! etc).
```

Comment:
Is this documentation moved somewhere else? Or is it not applicable anymore?

Reply:
Almost all is outdated.

Reply:
I think maybe the code example and the configuration explanation would be nice to keep

Reply:
I've added some documentation to `LOGGER`.